src/smoothing/SmoothingAndStep.java

```diff
@@ -80,6 +80,10 @@ public double smoothing(int[] params, String inPath, String outPath,
 
                 if((strLine = br.readLine()) != null){
                     while(!addLine(strLine)){
+                        if(Thread.currentThread().isInterrupted()){
```

Comment:
Indeed it feels suboptimal, and I'm trying to find a smoother way of doing it,  havent found anything yet though. From what I've read, it seems the only "solution" you can do is optimize the calls to `Thread.currentThread().isInterrupted()`. 

The problem I was having with the external process solution was either it didn't run the process at all, or it produced empty files.

However, this solves the problem, not elegantly, but it does solve it (as far as I can test, the rest is up to the researchers to check (validity of calculations and such)).